src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/TextDocumentExtensions.cs

```diff
@@ -25,7 +25,7 @@ public static bool TryComputeHintNameFromRazorDocument(this TextDocument razorDo
             return false;
         }
 
-        var projectBasePath = Path.GetDirectoryName(razorDocument.Project.FilePath);
+        var projectBasePath = Path.GetDirectoryName(razorDocument.Project.FilePath).AssumeNotNull();
         var relativeDocumentPath = razorDocument.FilePath[projectBasePath.Length..].TrimStart('/', '\\');
         hintName = RazorSourceGenerator.GetIdentifierFromPath(relativeDocumentPath);
```

Comment:
💭 Consider using spans here rather than allocating intermediate strings. I recognize that `RazorSourceGenerator.GetIdentifierFromPath` probably needs a look too, so I'm OK if you want to leave this as is.

Reply:
Made a note for the future. This is something I very much need, and want, to get better at